packages/browser-wallet/src/popup/pages/Account/Tokens/Tokens.tsx

```diff
@@ -0,0 +1,154 @@
+import { useSelectedCredential } from '@popup/shared/utils/account-helpers';
+import { TokenIdAndMetadata, WalletCredential } from '@shared/storage/types';
+import { tokenBalanceFamily, tokensAtom } from '@popup/store/token';
+import Button from '@popup/shared/Button';
+
+import { tokensRoutes } from './routes';
+
+type BalanceProps = {
+    atom: Atom<Promise<bigint>>;
+    decimals: number;
+};
+
+function Balance({ atom, decimals }: BalanceProps) {
+    const balance = useAtomValue(atom);
+    const getFraction = toFraction(10 ** decimals);
+    const renderBalance = (value: bigint) => addThousandSeparators(getFraction(value));
+
+    return <>{renderBalance(balance)}</>;
+}
+
+type FtProps = {
+    accountAddress: string;
+    contractIndex: string;
+    token: TokenIdAndMetadata;
+};
+
+function Ft({ accountAddress, contractIndex: contractAddress, token }: FtProps) {
+    const balanceAtom = tokenBalanceFamily(accountAddress, contractAddress, token.id);
+
+    return (
+        <Button clear className="token-list__item">
+            <img className="token-list__icon" src={token.metadata.thumbnail?.url} alt={token.metadata.name} />
+            {/* TODO should only show symbol for FTs, remove name fallback when NFTs work */}
+            <Suspense fallback={<>...</>}>
+                <Balance atom={balanceAtom} decimals={token.metadata.decimals ?? 0} />
+            </Suspense>{' '}
+            {token.metadata.symbol || token.metadata.name || ''}
+        </Button>
+    );
+}
+
+function useTokens(account: WalletCredential, unique: boolean) {
+    const {
+        loading,
+        value: { [account.address]: tokens },
+    } = useAtomValue(tokensAtom);
+    const items = useMemo(
+        () =>
+            loading || tokens === undefined
+                ? []
+                : Object.entries(tokens).flatMap(([contractIndex, ts]) => ts.map((t) => ({ contractIndex, ...t }))),
+        [tokens, loading]
+    );
+
+    return items.filter((t) => (t.metadata.unique ?? false) === unique);
+}
+
+type ListProps = {
+    account: WalletCredential;
+};
+
+function Fungibles({ account }: ListProps) {
+    const accountInfo = useAccountInfo(account);
+    const tokens = useTokens(account, false);
+
+    if (accountInfo === undefined) {
+        return null;
+    }
+
+    return (
+        <>
+            <Button clear className="token-list__item">
+                <CcdIcon className="token-list__icon token-list__icon--ccd" />
+                <div className="token-list__balance">{displayAsCcd(accountInfo.accountAmount)} CCD</div>
+            </Button>
+            {tokens.map((t) => (
+                <Ft
+                    key={`${t.contractIndex}.${t.id}`}
+                    accountAddress={account.address}
+                    contractIndex={t.contractIndex}
+                    token={t}
+                />
+            ))}
+        </>
+    );
+}
+
+function Collectibles({ account }: ListProps) {
```

Comment:
We should still have a way to tell the user if the collectible is owned or not?  (Maybe unowned has a more grey background or something? :thinking: )

Reply:
If you can fit "(ont onwed)" in there, then do it :+1: